tests/foreman/cli/test_capsulecontent.py

```diff
@@ -0,0 +1,206 @@
+            'environment-id': function_lce.id,
+        }
+    )
+    assert 'environment successfully added' in str(res)
+
+    # Check the capsule lists correct LCE, CV and repo names, but no content count before sync.
+    info = target_sat.cli.Capsule.content_info(
+        {'id': module_capsule_configured.nailgun_capsule.id, 'organization-id': function_org.id}
+    )
+    assert len(info['lifecycle-environments']) == 1, 'Too many or few LCEs listed'
+    lce_info = info['lifecycle-environments']['1']
+    assert lce_info['name'] == function_lce.name, 'Wrong LCE name listed'
+    assert lce_info['organization']['name'] == function_org.name, 'Wrong ORG name listed'
+    assert len(lce_info['content-views']) == 1, 'Too many or few CVs listed'
+    cv_info = lce_info['content-views']['1']
+    assert cv_info['name']['name'] == cv['name'], 'Wrong CV name listed'
+    assert len(cv_info['repositories']) == len(
+        cvv['repositories']
+    ), 'Too many or few repositories listed'
+    cv_info_reponames = set([repo['repository-name'] for repo in cv_info['repositories'].values()])
+    cvv_reponames = set([repo['name'] for repo in cvv['repositories']])
+    assert cv_info_reponames == cvv_reponames, 'Wrong repo names listed'
+    counts = [repo['content-counts'] for repo in cv_info['repositories'].values()]
+    assert all(
+        'Content view must be synced to see content counts' in i['warning'] for i in counts
+    ), 'Expected warning for all repo content counts'
+
+    # Sync, wait for counts to be updated and get them from the Capsule.
+    target_sat.cli.Capsule.content_synchronize(
+        {'id': module_capsule_configured.nailgun_capsule.id, 'organization-id': function_org.id}
+    )
+    target_sat.wait_for_tasks(
+        search_query='label = Actions::Katello::CapsuleContent::UpdateContentCounts',
+        search_rate=5,
+        max_tries=5,
+    )
+
+    # Get the content counts from Satellite side and compare them with Capsule.
+    info = target_sat.cli.Capsule.content_info(
+        {'id': module_capsule_configured.nailgun_capsule.id, 'organization-id': function_org.id}
+    )
+
+    caps_counts = [
+        repo
+        for repo in info['lifecycle-environments']['1']['content-views']['1'][
+            'repositories'
+        ].values()
+    ]
+    sat_repos = [
+        target_sat.cli.Repository.info({'id': repo['repository-id']}) for repo in caps_counts
+    ]
+
+    for s_repo in sat_repos:
+        c_repo = next(r for r in caps_counts if r['repository-id'] == s_repo['id'])
+        common_keys = set(s_repo['content-counts'].keys()) & set(c_repo['content-counts'].keys())
+        assert len(common_keys), f'''No common keys found for type "{s_repo['content-type']}".'''
+        assert expected_keys[s_repo['content-type']].issubset(common_keys), (
+            'Some fields are missing: expected '
+            f'''{expected_keys[s_repo['content-type']]} but found {common_keys}'''
```

Comment:
Why the triple quotes? How about `"`? I don't see triple quotes used like this anywhere in the repo.

Reply:
Indeed, no need for triples anymore.